addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+/**
+ * Declare a MapType where key is a string and value is an array of type string
+ * Use for storing a file's path as key and its dependent module's relative path
+ */
+declare type ImportMap {
+  [key: string]: ImportMap | string[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createtsSourceFile(fileName: string): ts.SourceFile {
+  return ts.createSourceFile(fileName,
+                             fs.readFileSync(fileName).toString(),
+                             ts.ScriptTarget.ES6,
+                             true);
+}
+
+/**
+ *  Store all the files' path under the given path in a passed in array.
+ *  All the files in the subdirectories of the given path are also included.
+ * 
+ * @param filePath path from where all the files' path are retrieved
+ * @param allFiles array where the files are stored and then returned.
+ * 
+ */
+
+export function getFiles(filePath: string, allFiles: string[]): any {
+  let files = fs.readdirSync(filePath);
+  files.forEach(file => {
+    if (path.extname(file) === '') {
+      getFiles(path.join(filePath, file), allFiles);
+    } else {
+      allFiles.push(path.join(filePath, file));
+    }
+  });
+  return allFiles;
+}
+
+/**
+ * Traverse through ts.Node recursively and stores moduleSpecifiers in an array
+ * 
+ * @param {ts.Node} node: Typescript Node of whose AST is being traversed
+ * @param {string[]} moduleSpecifiers: array to store moduleSpecifiers if found.
+ * 
+ */
+export function getModuleSpecifier(node: ts.Node, moduleSpecifiers: string[]): any {
```

Comment:
I notice you create a lot of functions with side effects. This is kind of frowned upon because people using your functions would probably not expect it. There's almost no advantage (surprisingly) of doing it that way rather than returning new objects, so in general we prefer to have pure functions.

To make this function pure, remove the moduleSpecifiers argument and create a local array. Then return that array at the end of the function (like you're already doing). When recursively calling `getModuleSpecifier`, instead do `moduleSpecifiers.push(...moduleSpecifiers(nodeObj));` or similar.